src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVCustomReadFilter.java

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Customized serializable reads filter, based on cmd line arguments provided, for structural variation purposes
+ */
+public final class SVCustomReadFilter implements ReadFilter {
+    private static final long serialVersionUID = 1L;
+
+    private static final PrimaryFilter pfilter = new PrimaryFilter();
+    private final CustomFilter sfilter;
+
+    public SVCustomReadFilter(final boolean[] sixFilterOptions,
+                              final CollectInsertSizeMetricsSpark.whichEndToUse whichEnd,
+                              final int mappingQualityThreshold
+                              //TODO: final int mateMappingQualityThreshold
+                             ){
+        sfilter = new CustomFilter(sixFilterOptions, whichEnd, mappingQualityThreshold);//TODO:, mateMappingQualityThreshold;
+    }
+    @Override
+    public boolean test(final GATKRead read){
+        return pfilter.and(sfilter).test(read);
+    }
+
+    /**
+     * Customary reads filter where filters are turned on/off by user options, which are passed in via constructor.
+     */
+    private static final class CustomFilter implements ReadFilter{
+        private static final long serialVersionUID = 1L;
+
+        private final int endVal;
+
+        private final ReadFilter collapsedFilter;
+
+        //TODO: private final int mateMQPassingThreshold = 0;
+
+        public CustomFilter(final boolean sixOptions[], final CollectInsertSizeMetricsSpark.whichEndToUse whichEnd,
+                            final int MQThreshold//, mateMQThreshold
+                            ){
+
+            endVal = whichEnd.getValue();
+
+            ReadFilter tempFilter = read -> 0!=read.getFragmentLength();
+                       tempFilter = tempFilter.and(read -> read.isPaired());
+                       tempFilter = tempFilter.and(read -> endVal == (read.isFirstOfPair() ? 1 : 2));
+
+            if(sixOptions[0]) { tempFilter = tempFilter.and(read -> read.isProperlyPaired());}
```

Comment:
the `sixOptions` must be expanded into 6 different fields with meaningful names or a class with 6 fields with meaningful names